rplugin/node/nvim_typescript/lib/index.js

```diff
@@ -195,30 +199,32 @@ let TSHost = class TSHost {
             const renameArgs = yield this.getCommonData();
             const buffNum = yield this.nvim.call('bufnr', '%');
             const renameResults = yield this.client.renameSymbol(Object.assign({}, renameArgs, { findInComments: false, findInStrings: false }));
-            if (renameResults) {
-                if (renameResults.info.canRename) {
-                    let changeCount = 0;
-                    for (let fileLocation of renameResults.locs) {
-                        let defFile = fileLocation.file;
-                        yield this.nvim.command(`e! ${defFile}`);
-                        for (let rename of fileLocation.locs) {
-                            let { line, offset } = rename.start;
-                            let substitutions = `${line}substitute/\\%${offset}c${symbol}/${newName}/`;
-                            yield this.nvim.command(substitutions);
-                            //list: Array<{ filename: string; lnum: number; col: number; text: string }>,
-                            changedFiles.push({ filename: defFile, lnum: line, col: offset, text: `Replaced ${symbol} with ${newName}` });
-                            changeCount += 1;
-                        }
+            if (this.isRenameSuccess(renameResults.info)) {
```

Comment:
The previous code checked that `renameResults` was truthy.

In this version, if `renameResults` is `null` or `undefined` it will throw.

Reply:
Oops, I reviewed the JS but it's still true in the TS.

Reply:
if the function possibly returned null, wouldn't it be caught by the compiler?

Reply:
I guess it could if the declared return type of the `renameSymbol` was `Promise<T|null>` explicitly, but it's not.

- The stack and return result of `_makeTssRequest` is non trivial
- The code _did_ check for null before

So I would leave the check as it was.

Reply:
ok I'll change it back there